subprojects/core/src/main/java/org/gradle/caching/internal/controller/BuildCacheLoadOp.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.controller;
+
+import org.gradle.caching.BuildCacheKey;
+
+import java.io.InputStream;
+
+/**
+ * A stateful record of a potential load operation.
+ */
+public interface BuildCacheLoadOp {
```

Comment:
I'll go with command.

We can't avoid the mutability because of the shape of the cache service API, without even more gymnastics.